apps/ledger-live-desktop/src/renderer/screens/asset/AssetBalanceSummaryHeader.tsx

```diff
@@ -71,23 +70,25 @@ export default function AssetBalanceSummaryHeader({
   }, [countervalueFirst, data]);
   const primaryKey = data[0].unit.code;
   const secondaryKey = data[1].unit.code;
-  const rampCatalog = useRampCatalog();
-  const availableOnBuy = useMemo(() => {
-    if (!rampCatalog.value) {
-      return false;
-    }
-    const allBuyableCryptoCurrencyIds = getAllSupportedCryptoCurrencyIds(rampCatalog.value.onRamp);
-    return allBuyableCryptoCurrencyIds.includes(currency.id);
-  }, [rampCatalog.value, currency.id]);
+  const { isCurrencyAvailable } = useRampCatalog();
+
+  const availableOnBuy = !!currency && isCurrencyAvailable(currency.id, "onRamp");
+
   const { providers, storedProviders } = useProviders();
   const startStakeFlow = useStakeFlow();
   const stakeProgramsFeatureFlag = useFeature("stakePrograms");
   const listFlag = stakeProgramsFeatureFlag?.params?.list ?? [];
   const stakeProgramsEnabled = stakeProgramsFeatureFlag?.enabled ?? false;
   const availableOnStake = stakeProgramsEnabled && currency && listFlag.includes(currency?.id);
-  const availableOnSwap = providers?.concat(storedProviders ?? []).some(({ pairs }) => {
-    return pairs && pairs.find(({ from, to }) => [from, to].includes(currency.id));
-  });
+
+  const availableOnSwap = providers
+    ?.concat(storedProviders ?? [])
+    .some(({ pairs }: { pairs: { from: string; to: string }[] }) => {
+      return (
+        pairs &&
+        pairs.find(({ from, to }: { from: string; to: string }) => [from, to].includes(currency.id))
```

Comment:
negligible improvement (so not necessary) but likely performance better if just comparing directly as `from === currency.id || to === curency.id` as you're creating a new array on each iteration